astrobee/survey_manager/survey_planner/src/survey_planner/command_astrobee.py

```diff
@@ -604,14 +605,16 @@ def move(self, from_name: str, to_name: str) -> int:
             to_name,
         )
         if exposure_value != 0:
-            cexec = CommandExecutor(self.cmd_exec_ns)
-            exit_code = first_non_zero(exit_code, cexec.change_exposure(exposure_value))
+            exit_code = first_non_zero(
+                exit_code, self.command_executor.change_exposure(exposure_value)
+            )
```

Comment:
I had intentionally made this change to create a new `CommandExecutor` every time because I ran into trouble with another action that reused a `CommandExecutor` and I thought the problem might have to do with the `CommandExecutor` having been left in an inconsistent state somehow such that it failed on the second pass.

The logic of `CommandExecutor` state is hard enough to follow that I felt more confident discarding the old one and creating a new one every time. Is there a significant resource concern with doing it that way? (I think it's generally a good idea in terms of isolation.)

I also want to point out that this code is probably not getting tested unless you are really going out of your way to do that (running multi-module scenarios), so we don't really know if either the old or the new version is working.

Reply:
Interesting. If there is a bug regarding that, we should focus on fixing the bug instead.
Creating a CommandExecutor instance takes something like 3 seconds because it has to publish and wait for subscribers. It seems like bad practice introducing a lag this big 3 times in a row xD